models/issue_comment.go

```diff
@@ -1118,7 +1135,19 @@ func fetchCodeCommentsByReview(e Engine, issue *Issue, currentUser *User, review
 		return nil, err
 	}
 
+	n := 0
 	for _, comment := range comments {
+		if re, ok := reviews[comment.ReviewID]; ok && re != nil {
+			// If the review is pending only the author can see the comments (except if the review is set)
+			if review.ID == 0 && re.Type == ReviewTypePending &&
+				(currentUser == nil || currentUser.ID != re.ReviewerID) {
+				continue
+			}
+			comment.Review = re
+		}
+		comments[n] = comment
+		n++
```

Comment:
I'm new to golang. Is this an OK way to remove the pending reviews from the array? I got it from https://github.com/golang/go/wiki/SliceTricks#filter-in-place.

Reply:
This is fine unless the slice is very big initially, much smaller afterwards and likely to hang around for a long time, or very big with only a few removals

* if you think the slice is going to hang around a long time and is quite big initially and small afterwards you should allocate a new a slice of the correct size at the end and issue `copy(filtered, slice)`

* If the slice is very big and is going to have only a few removals you should do: 
```go
	i := 0
	for i < len(slice) {
 		if !removing {
			i++
			continue
		}
		copy(slice[i:], slice[i+1:])
		slice= slice[:len(slice)-1]   
 	}
```

if the slice is expected to be less than 10, as it is,  then this is fine